compiler/rustc_hir_analysis/src/coherence/orphan.rs

```diff
@@ -220,16 +226,20 @@ fn do_orphan_check_impl<'tcx>(
             | ty::Ref(..)
             | ty::FnDef(..)
             | ty::FnPtr(..)
-            | ty::Closure(..)
+            | ty::Never
+            | ty::Tuple(..) => (LocalImpl::Allow, NonlocalImpl::DisallowOther),
+
+            ty::Closure(..)
             | ty::Generator(..)
             | ty::GeneratorWitness(..)
-            | ty::Never
-            | ty::Tuple(..)
-            | ty::Alias(AliasKind::Opaque, ..)
             | ty::Bound(..)
             | ty::Placeholder(..)
-            | ty::Infer(..)
-            | ty::Error(..) => (LocalImpl::Allow, NonlocalImpl::DisallowOther),
+            | ty::Infer(..) => {
+                tcx.sess.delay_span_bug(sp, "weird self type for autotrait impl");
```

Comment:
that should actually be possible as `span_bug!`, but it doesn't matter, change it if you have to rebase anyway or sth

Reply:
:+1: Will do.

I confirmed that `impl AutoTrait for _ {}` doesn't end up hitting the Infer or Placeholder case here. I guess something rewrites it to `impl<T> AutoTrait for T {}` by this point.